### PR TITLE
[Run2_2017] Cleanup Unused Branches

### PR DESCRIPTION
**Pull request by aperloff**
This PR removed lots of branches which are unused. We performed a survey of the analysis groups making use of the TreeMaker ntuples (https://docs.google.com/spreadsheets/d/1GjJXiwgOyQVE5TfFjsT1icxBNPyIwRsNshZWkVqNqog/edit?usp=sharing) and removed any branch not used by an analysis group. The size of the ntuples went from 6969.00 bytes/evt to 5570.00 bytes/evt.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -234,6 +234,8 @@ Brief explanation of the options in [makeTree.py](./TreeMaker/python/makeTree.py
 * `debugsubjets`: store some additional information for the AK8 subjets (default=False)
 * `applybaseline`: switch to apply the baseline HT selection (default=False)
 * `saveMinimalGenParticles`: save only the hard scatter gen particles coming from top decays, boson decays, semi-visible jets, or SUSY particles (default=True)
+* `saveGenTops`: save the 4-vectos of the generated tops and the TTbar reweighting scale factor (default=False)
```
Comment (kpedro88) [inline README.md]: typo: vectos -> vectors

File: TreeMaker/python/makeJetVars.py [hunk 2]
```diff
@@ -15,16 +15,17 @@ def makeMHTVars(self, process, JetTag, HTJetsTag, storeProperties, suff, MHTsuff
...
     
     from TreeMaker.Utils.mhtdouble_cfi import mhtdouble
     MHT = mhtdouble.clone(
         JetTag  = MHTJetsTag,
     )
     setattr(process,"MHT"+suff+MHTsuff,MHT)
-    self.VarsDouble.extend(['MHT'+suff+MHTsuff+':Pt(MHT'+suff+MHTsuff+')','MHT'+suff+MHTsuff+':Phi(MHTPhi'+suff+MHTsuff+')'])
-    
+    self.VarsDouble.extend(['MHT'+suff+MHTsuff+':Pt(MHT'+suff+MHTsuff+')'])
+    if storeProperties!=1: self.VarsDouble.extend(['MHT'+suff+MHTsuff+':Phi(MHTPhi'+suff+MHTsuff+')'])
```
Comment (kpedro88) [inline TreeMaker/python/makeJetVars.py]: can we put this back for all cases

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py [hunk 3]
```diff
@@ -264,8 +265,8 @@ def makeTreeFromMiniAOD(self,process):
         ak4updates = cms.PSet(discrs = cms.vstring())
         TMeras.TM80X.toModify(ak4updates,
             discrs = cms.vstring(
-                ['pfDeepCSVJetTags:'+x for x in ['probb','probc','probudsg','probbb']] +
-                ['pfDeepCSVDiscriminatorsJetTags:'+x for x in ['BvsAll','CvsB','CvsL']]
+                ['pfDeepCSVJetTags:'+x for x in ['probb','probbb']] +
```
Comment (kpedro88) [inline TreeMaker/python/makeTreeFromMiniAOD_cff.py:268]: follow up if we actually need these (and not the others)

**Review by kpedro88 — COMMENTED** [review]
Additional comments:
* remove the code in https://github.com/TreeMaker/TreeMaker/blob/Run2_2017/Utils/src/BasicSubstructureProducer.cc for ptdrlog and momenthalf (saves CPU)
* remove the following line from setup as we are no longer storing those branches:
https://github.com/TreeMaker/TreeMaker/blob/c3ec64233125ee55fc76b613c637944f88145648/setup.sh#L105

**Comment by kpedro88** [comment]
also remove:
https://github.com/TreeMaker/TreeMaker/blob/c3ec64233125ee55fc76b613c637944f88145648/setup.sh#L112-L115

**Comment by kpedro88** [comment]
and:
https://github.com/TreeMaker/TreeMaker/blob/c3ec64233125ee55fc76b613c637944f88145648/TreeMaker/python/makeTreeFromMiniAOD_cff.py#L454-L460
https://github.com/TreeMaker/TreeMaker/blob/c3ec64233125ee55fc76b613c637944f88145648/TreeMaker/python/makeTreeFromMiniAOD_cff.py#L497-L500

File: README.md (hunk 1 above)
Comment (aperloff) [reply README.md]: Fixed

File: TreeMaker/python/makeJetVars.py (hunk 2 above)
Comment (aperloff) [reply TreeMaker/python/makeJetVars.py]: Done

File: TreeMaker/python/makeTreeFromMiniAOD_cff.py (hunk 3 above)
Comment (aperloff) [reply TreeMaker/python/makeTreeFromMiniAOD_cff.py:268]: These have also been removed.

**Comment by aperloff** [comment]
I have removed the EgammaPostRecoTools parts in setup.sh and makeTreeFromMiniAOD.py.

**Comment by aperloff** [comment]
Are you sure I should have removed the NjettinessAxis_1027 line from the setup file? I checked out an entirely new release and ran the same test, but this time it had an error:
```
 ----- Begin Fatal Exception 26-Feb-2020 18:33:42 CST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 40 event: 30762 stream: 0
   [1] Running path 'WriteTree'
   [2] Prefetching for module TreeMaker/'TreeMaker2'
   [3] Prefetching for module JetProperties/'JetPropertiesAK8'
   [4] Calling method for module PATJetUpdater/'GoodJetsAK8Auxiliary'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edm::ValueMap<float>
Looking for module label: NjettinessBeta1AK8
Looking for productInstanceName: tau1etaAxis1

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

File: TreeMaker/python/makeJetVars.py [hunk 4]
```diff
@@ -325,8 +279,6 @@ def makeJetVarsAK8(self, process, JetTag, suff, storeProperties, SkipTag=cms.VIn
         setattr(process,"QGTagger"+suff,QGTagger)
         ak8floats.extend([
             'BasicSubstructure'+suff+':girth',
-            'BasicSubstructure'+suff+':momenthalf',
-            'BasicSubstructure'+suff+':ptdrlog',
             'NjettinessBeta1'+suff+':tau1etaAxis1',
```
Comment (kpedro88) [inline TreeMaker/python/makeJetVars.py]: remove these four lines too, that should fix the exception

Comment (aperloff) [reply TreeMaker/python/makeJetVars.py]: Now I get
```
----- Begin Fatal Exception 27-Feb-2020 10:22:51 CST-----------------------
An exception of category 'UnknownUserFloat' occurred while
   [0] Processing  Event run: 1 lumi: 40 event: 30762 stream: 0
   [1] Running path 'WriteTree'
   [2] Prefetching for module TreeMaker/'TreeMaker2'
   [3] Calling method for module JetProperties/'JetPropertiesAK8'
Exception Message:
Requested UserFloat BasicSubstructureAK8:girth is not available! Possible UserFloats are:
NjettinessAK8Puppi:tau1 NjettinessAK8Puppi:tau2 NjettinessAK8Puppi:tau3 NjettinessAK8Puppi:tau4 QGTaggerAK8:axis1 QGTaggerAK8:axis2 QGTaggerAK8:ptD ak8PFJetsCHSValueMap:NjettinessAK8CHSTau1 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau2 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau3 ak8PFJetsCHSValueMap:NjettinessAK8CHSTau4 ak8PFJetsCHSValueMap:ak8PFJetsCHSPrunedMass ak8PFJetsCHSValueMap:ak8PFJetsCHSSoftDropMass ak8PFJetsCHSValueMap:eta ak8PFJetsCHSValueMap:mass ak8PFJetsCHSValueMap:phi ak8PFJetsCHSValueMap:pt ak8PFJetsPuppiSoftDropMass ak8PFJetsPuppiSoftDropValueMap:nb1AK8PuppiSoftDropN2 ak8PFJetsPuppiSoftDropValueMap:nb1AK8PuppiSoftDropN3 ak8PFJetsPuppiSoftDropValueMap:nb2AK8PuppiSoftDropN2 ak8PFJetsPuppiSoftDropValueMap:nb2AK8PuppiSoftDropN3 jerFactor nb1AK8PuppiSoftDrop:ecfN2 nb1AK8PuppiSoftDrop:ecfN3 nb2AK8PuppiSoftDrop:ecfN2 nb2AK8PuppiSoftDrop:ecfN3 puppiSpecificAK8:HFEMPuppiMultiplicity puppiSpecificAK8:HFHadronPuppiMultiplicity puppiSpecificAK8:neutralHadronPuppiMultiplicity puppiSpecificAK8:neutralPuppiMultiplicity puppiSpecificAK8:photonPuppiMultiplicity puppiSpecificAK8:puppiMultiplicity updatedPatJetsTransientCorrectedAK8UpdatedJECAuxiliaryJECup updatedPatJetsTransientCorrectedAK8UpdatedJECAuxiliaryJERdown updatedPatJetsTransientCorrectedAK8UpdatedJECAuxiliaryJERup
----- End Fatal Exception -------------------------------------------------
```

File: TreeMaker/python/makeJetVars.py [hunk 5]
```diff
@@ -324,13 +278,6 @@ def makeJetVarsAK8(self, process, JetTag, suff, storeProperties, SkipTag=cms.VIn
         )
         setattr(process,"QGTagger"+suff,QGTagger)
         ak8floats.extend([
-            'BasicSubstructure'+suff+':girth',
```
Comment (kpedro88) [inline TreeMaker/python/makeJetVars.py:327]: this line should not be removed

**Comment by aperloff** [comment]
Checked out a new release and this PR still worked.

**Pull request merged**